contributing/chestertons-fence.md

```diff
@@ -0,0 +1,79 @@
+Exercism is a very intentional product - things are there because they've been designed to be there, and things are often left out because they've been designed to be left out.
+
+That said, every day someone comes up with a great idea that we've not thought of before - an idea that's new or better, or unblocks our thinking.
+So if you have an idea or a suggestion, we welcome it!
+
+But...
+
+Before posting it, please ask yourself whether we might have already considered it, and why we might have done things differently to what you're suggesting.
+And remember, the more obvious your idea seems - the more likely it's been discussed and debated many times over -- so when you present it, present it with the potential caveats and pitfalls in place.
+If you have an instinct to say "why not **just** ...", then it almost certainly falls into this category.
+
+## Do you have an example?
+
+A common frustration is that students submit code to mentors that does not pass the tests.
+An equally common suggestion is "Why not just automatically run the tests in the CLI before someone submits their code?"
+It seems like a great idea - the CLI can just call a command to run the tests, and not let the student submit if they're broken.
+
+So firstly, what does it mean to "just call a command to run the tests".
+It means writing a script for each of the 52 languages on Exercism that can run the tests and check the outcome.
+That's a bit of effort, but doable.
+
+But it also means writing that script so that it runs on Windows, MacOSX and Linux --  in any and all possible versions/configurations. That's a lot (if not an unbounded) amount of work.
+"Why does it have to be _every_ configuration?" you may ask?
+Well, because you're actively blocking someone from using Exercism unless they can run this script.
+That also means that it can never fail.
+If for some reason the script doesn't run, then the student is entirely blocked.
+A bug in one of those 52*3*n scripts means a student can't use the track any more on that OS.
+How do you test that?
+You can't.
+
+But, you say, you could just add a `--skip-tests` flag.
+Yes, you could - solid suggestion.
+But then, that means people can also choose to bypass the tests whenever they want anyway, which gets you back to the starting point, except now it's marginally less likely that broken tests are submitted, which means mentors have **less** expectation of that, meaning it's more jarring and confusing when the tests **are** broken.
+
+"But people wouldn't do that generally" you may argue. True, for some languages where it takes 0.5s to run the tests.
+But for languages that take 20s to run the tests, having to wait an extra 20s to submit is incredibly frustrating for students.
+
+So rather than posing the suggestion of "why not run the tests before submission", let's ask a question: "Why do people post solutions that don't pass the tests?"
```

Comment:
```suggestion
Regardless of where this conversation ends up, it's clear that there's a lot more complexity involved here than the simple premise of "just run the tests before someone submits". There are technical challenges to consider, workflow problems to think about, and the realisation that tracks differ widely enough something that is quick and easy for one is painful for another.

So rather than posing the suggestion of "why not run the tests before submission", let's ask a question: "Why do people post solutions that don't pass the tests?"
```